crates/pallet-executor/src/lib.rs

```diff
@@ -109,10 +117,37 @@ mod pallet {
 
             // TODO: ensure the receipt is ready to be applied
 
-            // TODO: apply the execution receipt.
+            let SignedExecutionReceipt {
+                execution_receipt, ..
+            } = signed_execution_receipt;
 
-            Self::deposit_event(Event::ExecutionReceiptStored {
-                receipt_hash: signed_execution_receipt.hash(),
+            let primary_hash = execution_receipt.primary_hash;
+            let primary_number = execution_receipt.primary_number;
+
+            // Execution receipt starts from the primary block #1.
+            if primary_number > 1u32 {
```

Comment:
Also check that this ER corresponds to `primary_number` that is either next after newest known before or older than that is missing in `ValidateUnsigned::validate_execution_receipt()`. If it is `>= best_primary_number + 1` it means we have missed some ERs and it is a bug.

Reply:
We'll handle the out-of-order issue in a later PR.